File: app/src/main/java/ir/goldenmind/advancedandroid/retrofit/IMDBRequestInterface.java
package ir.goldenmind.advancedandroid.retrofit;

import ir.goldenmind.advancedandroid.imdb.IMDBResponse;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface IMDBRequestInterface {
    @GET("/")
    Call<IMDBResponse> getMovieInfo(@Query("t") String movieName, @Query("apikey") String apiKey);
}

File: app/src/main/java/ir/goldenmind/advancedandroid/kotlinversion/retrofit/IMDBRequestInterface.kt
package ir.goldenmind.advancedandroid.kotlinversion.retrofit

import io.reactivex.Observable
import ir.goldenmind.advancedandroid.javaversion.imdb.IMDBResponse
import retrofit2.Response
import retrofit2.http.GET
import retrofit2.http.Query

interface IMDBRequestInterface {
    @GET("/")

    //Use Retofit without RxJava
    //fun getMovieInfo(@Query("t") movieName: String, @Query("apikey") apiKey: String) : Call<IMDBResponse>

    //Use Retofit with RxJava
    fun getMovieInfo(@Query("t") movieName: String, @Query("apikey") apiKey: String): Observable<IMDBResponse>

}